src/main/java/org/terasology/climateConditions/alterationEffect/BiomeSpecificSpeedAlterationSystem.java

```diff
@@ -0,0 +1,39 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.climateConditions.alterationEffect;
+
+import org.terasology.biomesAPI.Biome;
+import org.terasology.biomesAPI.BiomeRegistry;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.characters.GetMaxSpeedEvent;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.registry.In;
+
+import java.util.Optional;
+
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class BiomeSpecificSpeedAlterationSystem extends BaseComponentSystem {
+    @In
+    BiomeRegistry biomeRegistry;
+
+    @ReceiveEvent
+    public void modifySpeed(GetMaxSpeedEvent event, EntityRef entityRef, LocationComponent location) {
+        BiomeWalkSpeedComponent biomeWalkSpeed = entityRef.getComponent(BiomeWalkSpeedComponent.class);
+        if (biomeWalkSpeed != null) {
+            if (biomeWalkSpeed.biomeId.equals("")) {
+                event.multiply(biomeWalkSpeed.multiplier);
+            } else {
+                Optional<Biome> biome = biomeRegistry.getBiome(new Vector3i(location.getLocalPosition()));
+                if (biomeWalkSpeed.biomeId.equals(biome.get().getId())) {
```

Comment:
I haven't yet implemented this effect completely . It is important to know if this approach is acceptable. i.e. Is checking the biome in this way and modifying speed alright?

Reply:
Sorry if I missed something again, but what exactly does biome-specific walk speed have to do with adding the `BodyTemperatureAlterationEffect`? :thinking:

Reply:
It doesn't, but I need to know if implementing a biome Specific Walk speed effect this way is alright.., I'll shift this to a different PR if you think this way is fine...